src/server/htmlServer.ts

```diff
@@ -120,7 +126,10 @@ function getSuggestions(rowIndex: number, colIndex: number, fileUrl: URL) {
         } catch (e) {
             // Add telemetry log. Failed to parse liquid expression. (This may bloat up the logs so double check about this)
         }
+        autoCompleteTelemetryData.addMeasurement(TelemetryConstants.TIME_TAKEN_TO_PARSE, new Date().getTime() - timeStampBeforeLiquidParsing);
```

Comment:
yeah, you're thinking of C# kind of OO development. It's not needed in TS; just use typed interfaces and you'll reduce the complexity and generated code/perf costs.